contracts/integrations/test/internal-integration-tests/exchange_wrapper_test.ts

```diff
@@ -0,0 +1,1003 @@
+                        simulateFill(signedOrder, expectedFillResults, shouldPayWethFees);
+                    } else {
+                        totalFillResults.push(nullFillResults);
+                    }
+                });
+
+                const args: BatchFillArgs = [
+                    signedOrders,
+                    takerAssetFillAmounts,
+                    signedOrders.map(order => order.signature),
+                    {
+                        from: taker,
+                        gasPrice: DeploymentManager.gasPrice,
+                        value,
+                    },
+                ];
+
+                const fillResults = await deployment.exchange.batchFillOrdersNoThrow.callAsync(...args);
+                const receipt = await deployment.exchange.batchFillOrdersNoThrow.awaitTransactionSuccessAsync(...args);
+
+                expect(totalFillResults).to.be.deep.eq(fillResults);
+
+                await assertResultsAsync(receipt, fillTestInfo);
+            }
+
+            it('should transfer the correct amounts if all fees are paid in eth', async () => {
+                await testBatchFillOrdersNoThrowAsync(
+                    signedOrders.map(signedOrder => {
+                        return { signedOrder, isValid: true };
+                    }),
+                    DeploymentManager.protocolFee.times(signedOrders.length),
+                );
+            });
+
+            it('should transfer the correct amounts if some fees are paid in eth and some in weth', async () => {
+                await testBatchFillOrdersNoThrowAsync(
+                    signedOrders.map(signedOrder => {
+                        return { signedOrder, isValid: true };
+                    }),
+                    DeploymentManager.protocolFee.times(signedOrders.length - 1),
+                );
+            });
+
+            it('should transfer the correct amounts if all fees are paid in weth', async () => {
+                await testBatchFillOrdersNoThrowAsync(
+                    signedOrders.map(signedOrder => {
+                        return { signedOrder, isValid: true };
+                    }),
+                    constants.ZERO_AMOUNT,
+                );
+            });
+
+            it('should not revert if an order is invalid and fill the remaining orders', async () => {
+                const invalidOrder: SignedOrder = {
+                    ...signedOrders[0],
+                    signature: '0x00',
+                };
+
+                const validOrders = signedOrders.slice(1);
+                const newOrders = [invalidOrder, ...validOrders];
+
+                const takerAssetFillAmounts: BigNumber[] = [invalidOrder.takerAssetAmount.div(2)];
+
+                await testBatchFillOrdersNoThrowAsync(
+                    newOrders.map(signedOrder => {
+                        return { signedOrder, isValid: true };
```

Comment:
Why does this work? Shouldn't the first order have `isValid: false`?

Reply:
That's a really good point... Looking into this now, because it doesn't work if the "invalid" order has `isValid: false`.

Reply:
Turns out that I was using `signedOrders` in `testBatchFillOrdersNoThrowAsync` instead of `signedOrdersWithValidity`